src/Compiler/Driver/CompilerDiagnostics.fs

```diff
@@ -381,943 +376,1392 @@ let IsWarningOrInfoEnabled (diagnostic, severity) n level specificWarnOn =
     let rec SplitRelatedException exn =
-      match exn with
-      | ErrorFromAddingTypeEquation(g, denv, ty1, ty2, exn2, m) ->
-          let diag2, related = SplitRelatedException exn2
-          ErrorFromAddingTypeEquation(g, denv, ty1, ty2, diag2.Exception, m) |> ToPhased, related
-      | ErrorFromApplyingDefault(g, denv, tp, defaultType, exn2, m) ->
-          let diag2, related = SplitRelatedException exn2
-          ErrorFromApplyingDefault(g, denv, tp, defaultType, diag2.Exception, m) |> ToPhased, related
-      | ErrorsFromAddingSubsumptionConstraint(g, denv, ty1, ty2, exn2, contextInfo, m) ->
-          let diag2, related = SplitRelatedException exn2
-          ErrorsFromAddingSubsumptionConstraint(g, denv, ty1, ty2, diag2.Exception, contextInfo, m) |> ToPhased, related
-      | ErrorFromAddingConstraint(x, exn2, m) ->
-          let diag2, related = SplitRelatedException exn2
-          ErrorFromAddingConstraint(x, diag2.Exception, m) |> ToPhased, related
-      | WrappedError (exn2, m) ->
-          let diag2, related = SplitRelatedException exn2
-          WrappedError(diag2.Exception, m) |> ToPhased, related
-      // Strip TargetInvocationException wrappers
-      | :? TargetInvocationException as exn ->
-          SplitRelatedException exn.InnerException
-      | _ ->
-           ToPhased exn, []
-    SplitRelatedException diagnostic.Exception
+        match exn with
+        | ErrorFromAddingTypeEquation (g, denv, ty1, ty2, exn2, m) ->
+            let diag2, related = SplitRelatedException exn2
+            ErrorFromAddingTypeEquation(g, denv, ty1, ty2, diag2.Exception, m) |> ToPhased, related
+        | ErrorFromApplyingDefault (g, denv, tp, defaultType, exn2, m) ->
+            let diag2, related = SplitRelatedException exn2
+
+            ErrorFromApplyingDefault(g, denv, tp, defaultType, diag2.Exception, m)
+            |> ToPhased,
+            related
+        | ErrorsFromAddingSubsumptionConstraint (g, denv, ty1, ty2, exn2, contextInfo, m) ->
+            let diag2, related = SplitRelatedException exn2
+
+            ErrorsFromAddingSubsumptionConstraint(g, denv, ty1, ty2, diag2.Exception, contextInfo, m)
+            |> ToPhased,
+            related
+        | ErrorFromAddingConstraint (x, exn2, m) ->
+            let diag2, related = SplitRelatedException exn2
+            ErrorFromAddingConstraint(x, diag2.Exception, m) |> ToPhased, related
+        | WrappedError (exn2, m) ->
+            let diag2, related = SplitRelatedException exn2
+            WrappedError(diag2.Exception, m) |> ToPhased, related
+        // Strip TargetInvocationException wrappers
+        | :? TargetInvocationException as exn -> SplitRelatedException exn.InnerException
+        | _ -> ToPhased exn, []
 
+    SplitRelatedException diagnostic.Exception
 
 let DeclareMessage = DeclareResourceString
 
 do FSComp.SR.RunStartupValidation()
-let SeeAlsoE() = DeclareResourceString("SeeAlso", "%s")
```

Comment:
These changes are unfortunate.  We could increase `fsharp_max_function_binding_width` to make more of them format as single line, but that generally makes other code worse.  

Alternatively we could rid ourselves of all of these and move these strings to FSComp.txt, which doesn't need this insanity